Backend/src/Trackable.Services/Services/DispatchingService.cs

```diff
@@ -3,18 +3,25 @@
 
 using Microsoft.Extensions.Configuration;
 using Newtonsoft.Json.Linq;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Linq;
 using System.Net.Http;
 using System.Text;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
 using Trackable.Common.Exceptions;
 using Trackable.Models;
+using Trackable.Repositories;
+
 
 namespace Trackable.Services
 {
-    class DispatchingService : IDispatchingService
+    class DispatchingService : CrudServiceBase<int, Dispatch, IDispatchingRepository>, IDispatchingService
     {
+
+        public static ConcurrentDictionary<string, string> dictionary = new ConcurrentDictionary<string, string>();
```

Comment:
use a centralized Redis cache instead of local dictionary to sync between cloud service multiple instances.